apollo-testing-support/src/commonMain/kotlin/com/apollographql/apollo3/testing/internal/runTest.kt

```diff
@@ -0,0 +1,40 @@
+package com.apollographql.apollo3.testing.internal
+
+import com.apollographql.apollo3.annotations.ApolloInternal
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Utility method that calls [kotlinx.coroutines.test.runTest], with optional [before] and [after] blocks.
+ */
+@ApolloInternal
+@OptIn(ExperimentalCoroutinesApi::class)
+fun runTest(
+    context: CoroutineContext = EmptyCoroutineContext,
+    before: suspend CoroutineScope.() -> Unit = {},
+    after: suspend CoroutineScope.() -> Unit = {},
+    block: suspend CoroutineScope.() -> Unit,
+) {
+  kotlinx.coroutines.test.runTest(context) {
+    before()
+    try {
+      block()
+    } finally {
+      after()
+    }
+  }
+}
+
+/**
+ * Similar to [runTest] but uses `runBlocking` instead of [kotlinx.coroutines.test.runTest].
+ * Use it when delays/timing are important to the test or [kotlinx.coroutines.test.runTest] can't otherwise be used.
+ */
+@ApolloInternal
+expect fun runTestBlocking(
+    context: CoroutineContext = EmptyCoroutineContext,
+    before: suspend CoroutineScope.() -> Unit = {},
+    after: suspend CoroutineScope.() -> Unit = {},
+    block: suspend CoroutineScope.() -> Unit,
```

Comment:
Can we merge both? 

```kotlin
@ApolloInternal
@OptIn(ExperimentalCoroutinesApi::class)
fun runTest(
    skipDelays: Boolean = false, 
    before: suspend CoroutineScope.() -> Unit = {},
    after: suspend CoroutineScope.() -> Unit = {},
    block: suspend CoroutineScope.() -> Unit,
) {
  // I think using Dispatchers.Main will override the default `TestDispatcher` (but not sure)
  val context = if (skipDelays) EmptyCoroutinesContext else Dispatchers.Main
  kotlinx.coroutines.test.runTest(context) {
    before()
    try {
      block()
    } finally {
      after()
    }
  }
}
```

This way we can get rid of our own `Promise` and expect/actual stuff?

Reply:
Damn, too bad. `expect`/`actual` it is then. Nit, I'd rename `runTestBlocking` to something more explicit like `runBlockingOrPromise`. It's not pretty but it's internal only. And JS is not blocking at all so it's more accurate.

Reply:
Agreed!